src/text-buffer.js

```diff
@@ -1927,16 +1927,29 @@ class TextBuffer {
       try {
         await this.buffer.save(destination, this.getEncoding())
       } catch (error) {
-        const canEscalate = process.platform === 'darwin' || process.platform === 'linux'
-        if (error.code === 'EACCES' && destination === filePath && canEscalate) {
+        if (error.code !== 'EACCES' || destination !== filePath) throw error
+
+        const isMacOrLinux = process.platform === 'darwin' || process.platform === 'linux'
+        const isWindows = process.platform === 'win32'
+        if (isMacOrLinux) {
           const fsAdmin = require('fs-admin')
           try {
             await this.buffer.save(fsAdmin.createWriteStream(filePath), this.getEncoding())
           } catch (_) {
             throw error
           }
-        } else {
-          throw error
+        } else if (isWindows) {
+          const winattr = require('winattr')
+          const attrs = winattr.getSync(filePath)
```

Comment:
Shouldn't this be async so it does not block the main thread?
```suggestion
          const attrs = await winattr.get(filePath)
```

Reply:
`winattr.get` does not return a promise, it expects a callback, so I can't use `await` there. 

Also, I'm not sure it's using an async API at all, so it could still be blocking. I opted to use the `sync` variants to avoid the callback hell :P given saving a hidden file will rarely be a bottle neck I think it's a fair use.